docs/lang/articles/contribution/dev_install.md

```diff
@@ -4,318 +4,506 @@ sidebar_position: 2
-       tar xvJf llvm-10.0.0.src.tar.xz
-       cd llvm-10.0.0.src
-       mkdir build
-       cd build
-       cmake .. -DLLVM_ENABLE_RTTI:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=OFF -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86;NVPTX" -DLLVM_ENABLE_ASSERTIONS=ON -DLLVM_ENABLE_TERMINFO=OFF
-       # If you are building on Apple M1, use -DLLVM_TARGETS_TO_BUILD="AArch64".
-       # If you are building on NVIDIA Jetson TX2, use -DLLVM_TARGETS_TO_BUILD="ARM;NVPTX"
-       # If you are building for a PyPI release, add -DLLVM_ENABLE_Z3_SOLVER=OFF to reduce the library dependency.
-       make -j 8
-       sudo make install
-       # Check your LLVM installation
-       llvm-config --version  # You should get 10.0.0
-       ```
-
-     - For Windows:
-
-       ```bash
-       # For Windows
-       # LLVM 10.0.0 + MSVC 2019
-       cmake .. -G "Visual Studio 16 2019" -A x64 -DLLVM_ENABLE_RTTI:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=OFF   -DCMAKE_BUILD_TYPE=Release -DLLVM_TARGETS_TO_BUILD="X86;NVPTX" -DLLVM_ENABLE_ASSERTIONS=ON -Thost=x64   -DLLVM_BUILD_TESTS:BOOL=OFF -DCMAKE_INSTALL_PREFIX=installed
-       ```
-
-       - Then open `LLVM.sln` and use Visual Studio 2017+ to build.
-       - Please make sure you are using the `Release` configuration.
-         After building the `INSTALL` project (under folder
-         `CMakePredefinedTargets` in the Solution Explorer window).
-       - If you use MSVC 2019, **make sure you use C++17** for the
-         `INSTALL` project.
-       - After the build is complete, find your LLVM binaries and
-         headers in `build/installed`.
-       - Please add `build/installed/bin` to `PATH`. Later, when you build Taichi, please use `cmake -DLLVM_DIR=<path_to_build>/build/installed/lib/cmake/llvm`.
-
-
-### Setting up CUDA (optional)
+<TabItem value="macos">
 
-:::note
-To build with NVIDIA GPU support, CUDA 10.0+ is needed. This installation guide works for Ubuntu 16.04+.
-:::
+1. Ensure that the Clang that ships with your MacBook has a version &ge;8 and &lt;12:
 
-If you don't have CUDA, go to [this website](https://developer.nvidia.com/cuda-downloads) and download the installer.
+  ```
+  clang --version
+  ```
 
-- To check if CUDA is installed, run `nvcc --version` or
-  `cat /usr/local/cuda/version.txt`.
-- On **Ubuntu** we recommend choosing `deb (local)` as **Installer
-  Type**.
-- On **Arch Linux**, you can easily install CUDA via `pacman -S cuda`
-  without downloading the installer manually.
+2. If your Clang version is &ge;12, install Clang 11:
 
-:::note
-If you are using a machine with an earlier CUDA version and/or old generation GPUs. We suggest to consult the [Compatibility Document](https://docs.nvidia.com/deploy/cuda-compatibility/) and the [CUDA Installation Guide](https://docs.nvidia.com/cuda/cuda-installation-guide-linux/index.html) first.
-:::
+  ```
+  brew install llvm@11
+  export CXX=/opt/homebrew/opt/llvm@11/bin/clang++
```

Comment:
Sorry to comment late. This line could be more solid by

```
export CXX=$(brew --prefix llvm@11)/bin/clang++
```

Because `/opt/homebrew/opt/` is not always the path but `brew --prefix llvm@11` always give the path if only you install by homebrew.